#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Fix typo in help message for caller builtin

--- bash/builtins/caller.def~	2006-08-24 17:30:39.000000000 +0200
+++ bash/builtins/caller.def	2008-04-19 23:28:44.000000000 +0200
@@ -130,7 +130,7 @@
 static char *caller_doc[] = {
   N_("Returns the context of the current subroutine call."),
   N_(" "),
-  N_("Without EXPR, returns returns \"$line $filename\".  With EXPR,"),
+  N_("Without EXPR, returns \"$line $filename\".  With EXPR,"),
   N_("returns \"$line $subroutine $filename\"; this extra information"),
   N_("can be used used to provide a stack trace."),
   N_(" "),
